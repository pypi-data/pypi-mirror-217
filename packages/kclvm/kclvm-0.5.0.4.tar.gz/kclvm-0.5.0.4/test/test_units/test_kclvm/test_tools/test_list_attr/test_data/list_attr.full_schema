FullName, attr:[firstName, lastName],
Person1, attr:[name, card, _mybkLabels, commands], parent:Person, attr:[name, age], parent:Persona, attr:[name, age],
Person2, attr:[card], parent:Person1, attr:[name, card, _mybkLabels, commands], parent:Person, attr:[name, age], parent:Persona, attr:[name, age],