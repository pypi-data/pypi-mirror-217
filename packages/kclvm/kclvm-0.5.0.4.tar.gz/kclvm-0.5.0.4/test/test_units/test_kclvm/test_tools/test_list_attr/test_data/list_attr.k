import import_file
# test for builtin
import math
# test for plugin
import kcl_plugin.app_context as ctx

schema FullName:
    firstName: str
    lastName: str

schema Person1(Person):
    name: FullName = FullName{
        firstName = "hello"
        lastName = "world"
    }
    card: str = "abc"
    _mybkLabels?: {str:str} = {
        if _workspace: "app.stack.io/workspace": _workspace.lower()
        "app.kubernetes.io/name": _appName
        "app.kubernetes.io/instance": _appName
    } if _globalTenantName == "CLOUDCORE" else Undefined
    commands: [str] = ["a", if True : "b"]

schema Person2(Person1):
    card: str
