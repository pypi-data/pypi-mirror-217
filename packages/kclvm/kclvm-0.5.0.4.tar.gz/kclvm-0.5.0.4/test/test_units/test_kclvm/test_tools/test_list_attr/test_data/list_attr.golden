------------ schema list ------------
Here are schemas defined in list_attr.k:
- FullName
- Person1
- Person2
Here are schemas imported to list_attr.k:
imported from import_file.k
- Person
imported from nested_import_file.k
- Persona
------------ schema structures ------------
schema FullName:
name                          type                          default                       is_final  is_optional
firstName                     str                                                                   Required  
lastName                      str                                                                   Required  

schema Person1:
name                          type                          default                       is_final  is_optional
name                          FullName                      FullName{...}                           Required  
card                          str                           "abc"                                   Required  
_mybkLabels                   {str:str}                     ...                                               
commands                      [str]                         ...                                     Required  
attrs inherited from Person
name                          str                                                                   Required  
age                           int                                                                   Required  
attrs inherited from Persona
name                          str                                                                   Required  
age                           int                                                                   Required  

schema Person2:
name                          type                          default                       is_final  is_optional
card                          str                                                                   Required  
attrs inherited from Person1
name                          FullName                      FullName{...}                           Required  
card                          str                           "abc"                                   Required  
_mybkLabels                   {str:str}                     ...                                               
commands                      [str]                         ...                                     Required  
attrs inherited from Person
name                          str                                                                   Required  
age                           int                                                                   Required  
attrs inherited from Persona
name                          str                                                                   Required  
age                           int                                                                   Required  

